core/network/space.go

```diff
@@ -14,16 +14,68 @@ import (
 
 	coreerrors "github.com/juju/juju/core/errors"
 	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/uuid"
 )
 
+// SpaceUUID represents a space unique identifier.
+type SpaceUUID string
+
+// NewID is a convenience function for generating a new space uuid.
+func NewSpaceUUID() (SpaceUUID, error) {
+	uuid, err := uuid.NewUUID()
+	if err != nil {
+		return SpaceUUID(""), err
+	}
+	return SpaceUUID(uuid.String()), nil
+}
+
+// ParseSpaceUUID returns a new SpaceUUID from the given string. If the string is not a valid
+// uuid an error satisfying [errors.NotValid] will be returned.
+func ParseSpaceUUID(value string) (SpaceUUID, error) {
+	if !uuid.IsValidUUIDString(value) {
+		return "", errors.Errorf("SpaceUUID %q %w", value, coreerrors.NotValid)
+	}
+	return SpaceUUID(value), nil
+}
+
+// String implements the stringer interface for SpaceUUID.
+func (u SpaceUUID) String() string {
+	return string(u)
+}
+
+// Validate ensures the consistency of the SpaceUUID. If the uuid is invalid an error
+// satisfying [errors.NotValid] will be returned.
+func (u SpaceUUID) Validate() error {
+	if u == "" {
+		return errors.Errorf("id cannot be empty").Add(coreerrors.NotValid)
+	}
+	if !uuid.IsValidUUIDString(string(u)) {
+		return errors.Errorf("id %q %w", u, coreerrors.NotValid)
+	}
+	return nil
+}
+
+// SpaceName represents a space name, used as a human-readable unique identifier.
+type SpaceName string
```

Comment:
It's a shame we can't call this `Name` and reference by package like we do with units. One day we'll split this whole network package.